src/Components/NavBar/Menus/NotificationsMenu.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { useContext } from "react"
+import { graphql, QueryRenderer } from "react-relay"
+
+import { SystemContext } from "Artsy"
+import { renderWithLoadProgress } from "Artsy/Relay/renderWithLoadProgress"
+import { get } from "Utils/get"
+
+import {
+  NotificationsMenuQuery,
+  NotificationsMenuQueryResponse,
+} from "__generated__/NotificationsMenuQuery.graphql"
+
+import {
+  Box,
+  Flex,
+  Image,
+  Link,
+  Menu,
+  MenuItem,
+  Sans,
+  Separator,
+} from "@artsy/palette"
+
+const NotificationsMenu: React.FC<NotificationsMenuQueryResponse> = props => {
+  const notifications = get(props, p => p.me.followsAndSaves.notifications.edges, []) // prettier-ignore
+
+  return (
+    <>
+      {notifications.map(({ node }, index) => {
+        const { artists, href, image, summary } = node
+        return (
+          <MenuItem href={href} key={index}>
+            <Flex alignItems="center">
+              <Box width={40} height={40} bg="black5" mr={1}>
+                <Image
+                  src={image.resized.url}
+                  width={40}
+                  height={40}
+                  style={{
+                    objectFit: "cover",
+                  }}
+                />
+              </Box>
+              <Box>
+                <Sans size="2">{summary}</Sans>
+                <Sans size="2" weight="medium">
+                  {artists}
+                </Sans>
+              </Box>
+            </Flex>
+          </MenuItem>
+        )
+      })}
+
+      <Flex pt={1} pb={1} flexDirection="column" alignItems="center">
+        {notifications.length ? (
+          <>
+            <Separator />
+            <Box pt={2}>
+              <Sans size="2">
+                <Link href="/works-for-you">View all</Link>
+              </Sans>
+            </Box>
+          </>
+        ) : (
+          <Box>
+            <Sans size="3">No new notifications</Sans>
+          </Box>
+        )}
+      </Flex>
+    </>
+  )
+}
+
+export const NotificationsMenuQueryRenderer = props => {
+  const { relayEnvironment } = useContext(SystemContext)
+
+  return (
+    <Menu title="Actvity">
+      <QueryRenderer<NotificationsMenuQuery>
+        environment={relayEnvironment}
+        query={graphql`
+          query NotificationsMenuQuery {
+            me {
+              followsAndSaves {
+                notifications: bundledArtworksByArtist(
+                  sort: PUBLISHED_AT_DESC
+                  for_sale: true
+                  first: 10
+                ) @connection(key: "WorksForYou_notifications") {
```

Comment:
@mzikherman - I grabbed this query from the `/works-for-you` page. I'm not 100% this is correct though because it doesn't quite match up with what's returned on the current `artsy.net`. Thoughts?

Reply:
Yup! The schema in https://github.com/artsy/metaphysics/blob/ce42aef4bd4767e66efe8baf49eb1d621cf3eafb/src/schema/me/notifications.ts should correspond to the drop-down. (It also has the read/unread indicator stuff).

Fundamentally though, it's all the same data- artworks by artists you follow, grouped by artist/date, in descending order of publish date. But the exact grouping across pagination and order of artists are different, so at first glance (especially for admins who have tons of followed artists) it for sure seems totally different.

Given that we link to /works-for-you from the dropdown, I think it's ok to use this, but you may want to explore the notifications one? It also has support for the read/unread indicator (which theoretically could use a mutation to mark it read as well).

Reply:
The 'total count' of these grouped notifications (for the overlay on the bell icon) isnt exposed yet in MP but is part of the notifications schema.